autoscale-groups-core/src/main/java/fish/payara/extensions/autoscale/groups/core/admin/ScaleCommandHelper.java

```diff
@@ -0,0 +1,327 @@
+                LOGGER.warning(msg);
+                failureOccurred = true;
+                // Re-establish interrupted state on thread
+                Thread.currentThread().interrupt();
+                break;
+            }
+
+            if (cmdRunnable == null) {
+                // We've timed out.
+                break;
+            }
+            String cname = cmdRunnable.getName();
+            waitingForServerNames.remove(cname);
+            ActionReport instanceReport = cmdRunnable.getActionReport();
+            if (LOGGER.isLoggable(FINE)) {
+                LOGGER.fine(String.format("Instance %d of %d (%s) has responded with %s", n + 1, nInstances, cname,
+                        instanceReport.getActionExitCode()));
+            }
+
+            if (instanceReport.getActionExitCode() != SUCCESS) {
+                // Bummer, the command had an error. Log and save output
+                failureOccurred = true;
+                failedServerNames.append(cname).append(" ");
+                reportResult.failedServerNames.add(cname);
+                String msg = cname + ": " + instanceReport.getMessage();
+                LOGGER.severe(msg);
+                msg = Strings.get("cluster.command.instancesFailed", commandName, cname);
+                progressStatus.progress(1, msg);
+            } else {
+                // Command worked. Note that too.
+                succeededServerNames.append(cname).append(" ");
+                reportResult.succeededServerNames.add(cname);
+                progressStatus.progress(1, cname);
+            }
+        }
+
+        actionReport.setActionExitCode(SUCCESS);
+
+        if (failureOccurred) {
+            actionReport.setResultType(List.class, reportResult.failedServerNames);
+        } else {
+            actionReport.setResultType(List.class, reportResult.succeededServerNames);
+        }
+
+        // Display summary of started servers
+        if (succeededServerNames.length() > 0) {
+            output.append("\n").append(Strings.get("cluster.command.instancesSucceeded", commandName,
+                    succeededServerNames));
+        }
+
+        if (failureOccurred) {
+            // Display summary of failed servers if we have any
+            output.append("\n").append(Strings.get("cluster.command.instancesFailed", commandName,
+                    failedServerNames));
+            if (succeededServerNames.length() > 0) {
+                // At least one instance started. Warning.
+                actionReport.setActionExitCode(WARNING);
+            } else {
+                // No instance started. Failure
+                actionReport.setActionExitCode(FAILURE);
+            }
+        }
+
+        // Check for server that did not respond
+        if (!waitingForServerNames.isEmpty()) {
+            String msg = Strings.get("cluster.command.instancesTimedOut", commandName,
```

Comment:
This may reference the wrong message string, see my previous comment

Reply:
`Timed out while waiting for command {0} to complete for: {1}`